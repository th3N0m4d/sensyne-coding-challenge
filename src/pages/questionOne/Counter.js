import React from 'react'
import Button from '@material-ui/core/Button'
import PropTypes from 'prop-types'

import { Container } from './atoms'

const propTypes = {
  onClick: PropTypes.func
}

const Counter = ({ count, onClick }) => (
  <Container>
    <Button variant='contained' onClick={onClick}>
      {`I've been clicked: ${count} times`}
    </Button>
  </Container>
)

Counter.propTypes = propTypes

export default Counter
