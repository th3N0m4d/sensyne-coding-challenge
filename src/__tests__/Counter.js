import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Button from '@material-ui/core/Button'

import Counter from '../pages/questionOne/Counter'

const expectToMatchSnapshot = component => {
  expect(renderer.create(component).toJSON()).toMatchSnapshot()
}

describe('Counter', () => {
  it('should render component', () => {
    expectToMatchSnapshot(<Counter />)
  })

  it('should render component with initial count of 7', () => {
    expectToMatchSnapshot(<Counter count={7} />)
  })

  it('should trigger onClick', () => {
    const onClickSpy = jest.fn()
    const wrapper = shallow(<Counter onClick={onClickSpy} />)

    const button = wrapper.find(Button)

    button.simulate('click')

    expect(onClickSpy).toHaveBeenCalled()
  })
})
