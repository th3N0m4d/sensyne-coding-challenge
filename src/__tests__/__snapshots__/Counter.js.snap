// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render component 1`] = `
<div
  className="sc-AxjAm jTNVWA"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      I've been clicked: undefined times
    </span>
  </button>
</div>
`;

exports[`Counter should render component with initial count of 7 1`] = `
<div
  className="sc-AxjAm jTNVWA"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      I've been clicked: 7 times
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
